data/render/src/lib.rs

```diff
@@ -21,6 +23,65 @@ pub fn validate(tags: &Tags, entries: &Vec<Entry>) -> Result<(), Box<dyn Error>>
     Ok(())
 }
 
+#[tokio::main]
+pub async fn check_deprecated(token: std::string::String, entries: &mut Vec<Entry>) -> Result<Vec<Entry>, Box<dyn Error>> {
+    let github = Github::new(
+        String::from("user-agent-name"),
+        Credentials::Token(token),
+    )?;
+
+    let mut entries_tmp: Vec<Entry> = entries.to_vec();
+    for entry in &mut entries_tmp {
+        if entry.source.is_none() {
+            continue;
+        }
+
+        let mut source: &str = entry.source.as_ref().unwrap();
+        if source.chars().last().unwrap() == '/' {
+            source = source.trim_end_matches('/');
+        }
```

Comment:
I edited like this, looks a bit long for the first line though

```
let components: Vec<&str> = entry.source.as_ref().unwrap().trim_end_matches('/').split("/").collect();
if !(components.contains(&"github.com") && components.len() == 5) {
    // valid github source must have 5 elements - anything longer and they are probably a
    // reference to a path inside a repo, rather than a repo itself.
    continue;
}
```

not sure if this make sense?

Reply:
You could refactor it a bit and put the extraction of `owner` and `repo` into a seprate function:

```rust
let (owner, repo) = analyze_repo_url(entry.source)?;
```

Code for the function:

```rust
fn analyze_repo_url(url: String) -> Result<(String, String), Box<dyn Error>> { 
  let components: Vec<&str> = url.trim_end_matches('/').split("/").collect();
  if !components.contains(&"github.com") || components.len() != 5 {
    // valid github source must have 5 elements - anything longer and they are probably a
    // reference to a path inside a repo, rather than a repo itself.
    return Err("Cannot extract owner and repo".into());
  }
  let owner = components[3];
  let repo  = components[4];
  Ok((owner.into(), repo.into()))
}
```

Completely optional, though. 😉